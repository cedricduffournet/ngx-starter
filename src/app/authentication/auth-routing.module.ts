import { NgModule } from '@angular/core';
import { Routes, RouterModule } from '@angular/router';
import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent
} from '@app/authentication/containers';
import { authConfig } from '@app/authentication/auth.config';

const AUTHENTICATION_ROUTES: Routes = [
  { path: 'login', component: LoginViewComponent },
  { path: 'resetting', component: SendResetPasswordViewComponent },
  {
    path: `${authConfig.path.passwordResetConfirm}/:token`,
    component: ResetPasswordViewComponent
  },
  { path: 'register', component: RegisterViewComponent },
  {
    path: `${authConfig.path.registerConfirm}/:token`,
    component: RegisterConfirmViewComponent
  }
];

@NgModule({
  imports: [RouterModule.forChild(AUTHENTICATION_ROUTES)],
  exports: [RouterModule]
})
export class AuthRoutingModule {}
