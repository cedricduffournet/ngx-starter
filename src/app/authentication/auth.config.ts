export const authConfig = {
  path: {
    registerConfirm: 'register/confirm',
    passwordResetConfirm: 'password/reset'
  },
  storageKey: 'authentication'
};
