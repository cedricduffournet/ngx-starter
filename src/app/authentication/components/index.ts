export * from './auth-layout/auth-layout.component';
export * from './login-form/login-form.component';
export * from './register-form/register-form.component';
export * from './reset-password-form/reset-password-form.component';
export * from './send-reset-password-form/send-reset-password-form.component';
