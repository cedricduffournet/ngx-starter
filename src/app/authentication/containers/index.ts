export * from './login-view/login-view.component';
export * from './register-confirm-view/register-confirm-view.component';
export * from './register-view/register-view.component';
export * from './reset-password-view/reset-password-view.component';
export * from './send-reset-password-view/send-reset-password-view.component';
export * from './logout-modal/logout-modal.component';
