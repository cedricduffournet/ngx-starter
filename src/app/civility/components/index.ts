export * from './civility-add/civility-add.component';
export * from './civility-update/civility-update.component';
export * from './civility-delete/civility-delete.component';
export * from './civility-form/civility-form.component';
export * from './civility-item/civility-item.component';
export * from './civility-items/civility-items.component';
