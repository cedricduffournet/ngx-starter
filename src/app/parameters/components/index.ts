export * from './referenced-table-menu/referenced-table-menu.component';
