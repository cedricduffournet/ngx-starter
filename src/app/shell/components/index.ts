export * from './navbar/navbar.component';
export * from './dropdown-navbar/dropdown-navbar.component';
export * from './menu/menu.component';
