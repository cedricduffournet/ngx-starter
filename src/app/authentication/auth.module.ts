import { NgModule } from '@angular/core';

import { StoreModule } from '@ngrx/store';
import { EffectsModule } from '@ngrx/effects';

import { SharedModule } from '@app/shared/shared.module';
import { CivilityModule } from '@app/civility/civility.module';
import { LangSwitcherModule } from '@app/shared/lang-switcher/lang-switcher.module';
import { LogoModule } from '@app/shared/logo/logo.module';
import { AuthRoutingModule } from '@app/authentication/auth-routing.module';
import { AuthEffects } from '@app/authentication/state/effects';
import { reducers } from '@app/authentication/state/reducers';
import { AuthFacade } from '@app/authentication/state/auth.facade';

import {
  LoginViewComponent,
  SendResetPasswordViewComponent,
  ResetPasswordViewComponent,
  RegisterViewComponent,
  RegisterConfirmViewComponent,
  LogoutModalComponent
} from '@app/authentication/containers';

import {
  AuthLayoutComponent,
  LoginFormComponent,
  ResetPasswordFormComponent,
  SendResetPasswordFormComponent,
  RegisterFormComponent
} from '@app/authentication/components';

@NgModule({
  imports: [
    AuthRoutingModule,
    SharedModule,
    StoreModule.forFeature('auth', reducers),
    EffectsModule.forFeature([AuthEffects]),
    LangSwitcherModule,
    LogoModule,
    CivilityModule
  ],
  declarations: [
    AuthLayoutComponent,
    LoginViewComponent,
    SendResetPasswordViewComponent,
    ResetPasswordViewComponent,
    RegisterViewComponent,
    RegisterConfirmViewComponent,
    LoginFormComponent,
    ResetPasswordFormComponent,
    SendResetPasswordFormComponent,
    LogoutModalComponent,
    RegisterFormComponent
  ],
  entryComponents: [LogoutModalComponent],
  providers: [AuthFacade]
})
export class AuthModule {}
