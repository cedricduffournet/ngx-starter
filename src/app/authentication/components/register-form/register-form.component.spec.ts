import { TestBed, ComponentFixture } from '@angular/core/testing';
import { ReactiveFormsModule } from '@angular/forms';
import { TranslateModule } from '@ngx-translate/core';

import { FormModule } from '@app/shared/form';
import { RegisterFormComponent } from '@app/authentication/components';

describe('RegisterForm', () => {
  let fixture: ComponentFixture<RegisterFormComponent>;
  let component: RegisterFormComponent;
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, TranslateModule.forRoot(), FormModule],
      declarations: [RegisterFormComponent]
    });
    fixture = TestBed.createComponent(RegisterFormComponent);
    component = fixture.componentInstance;
  });

  it('should be created', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should not emit event if civility is empty', () => {
    const register = {
      civility: null,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test.com',
      plainPassword: {
        first: 'password1',
        second: 'password1'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if firstname is empty', () => {
    const register = {
      civility: 1,
      firstname: '',
      lastname: 'lastname',
      email: 'test@test.com',
      plainPassword: {
        first: 'password1',
        second: 'password1'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if lastname is empty', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: '',
      email: 'test@test.com',
      plainPassword: {
        first: 'password1',
        second: 'password1'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if email is empty', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: '',
      plainPassword: {
        first: 'password1',
        second: 'password1'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if email is invalid', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test',
      plainPassword: {
        first: 'password1',
        second: 'password1'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if password is empty', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test.fr',
      plainPassword: {
        first: '',
        second: ''
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if password does not match requirement', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test.fr',
      plainPassword: {
        first: 'aaa12',
        second: 'aaa12'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).not.toHaveBeenCalled();
  });

  it('should not emit event if password and password confirmation are different', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test.fr',
      plainPassword: {
        first: 'aaaA12',
        second: 'aaaB12'
      }
    };

    component.registerForm.setValue(register);
    fixture.detectChanges();
    spyOn(component.register, 'emit');
    component.onSubmit();
  });

  it('should emit register event if the form is valid when submitted', () => {
    const register = {
      civility: 1,
      firstname: 'firstname',
      lastname: 'lastname',
      email: 'test@test.fr',
      plainPassword: {
        first: 'password123',
        second: 'password123'
      }
    };

    component.registerForm.setValue(register);

    spyOn(component.register, 'emit');
    component.onSubmit();

    expect(component.register.emit).toHaveBeenCalled();
  });

  it('should disable submit button form if processing', () => {
    component.processing = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should display user exist message if provided', () => {
    component.errorUserExist = true;

    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
