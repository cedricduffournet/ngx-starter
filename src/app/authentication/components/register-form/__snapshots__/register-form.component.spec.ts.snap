// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm should be created 1`] = `
<app-register-form
  errorUserExist="false"
  processing="false"
  register={[Function EventEmitter]}
  submitted="false"
  userFormService={[Function UserFormService]}
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div
      class="form-group"
    >
      <label
        class="control-label"
        ng-reflect-translate=""
        translate=""
      >
        CIVILITY
      </label>
      <select
        class="form-control custom-select ng-untouched ng-pristine ng-invalid"
        formcontrolname="civility"
        ng-reflect-name="civility"
      >
        <option
          ng-reflect-translate=""
          translate=""
          value="0: null"
        >
          CHOOSE
        </option>
        
      </select>
    </div>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="firstname"
      ng-reflect-label="FIRST_NAME"
      ng-reflect-name="firstname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          FIRST_NAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="lastname"
      ng-reflect-label="LASTNAME"
      ng-reflect-name="lastname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LASTNAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="email"
      ng-reflect-error-text=""
      ng-reflect-label="EMAIL"
      ng-reflect-name="email"
      ng-reflect-show-error="false"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          EMAIL
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-row ng-untouched ng-pristine ng-invalid"
      formgroupname="plainPassword"
      ng-reflect-name="plainPassword"
    >
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="first"
          ng-reflect-help-text="HELP_PASSWORD_FIELD"
          ng-reflect-label="PASSWORD"
          ng-reflect-name="first"
          ng-reflect-submitted="false"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
            <small
              class="form-text text-muted"
            >
               HELP_PASSWORD_FIELD 
            </small>
          </div>
        </app-text-input>
      </div>
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-valid"
          formcontrolname="second"
          ng-reflect-error-text="FIELD_PASSWORD_MUST_BE_EQUAL"
          ng-reflect-label="PASSWORD_CONFIRMATION"
          ng-reflect-name="second"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD_CONFIRMATION
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
          </div>
        </app-text-input>
      </div>
    </div>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CREATE_ACCOUNT 
      </button>
    </div>
  </form>
</app-register-form>
`;

exports[`RegisterForm should disable submit button form if processing 1`] = `
<app-register-form
  errorUserExist="false"
  processing={[Function Boolean]}
  register={[Function EventEmitter]}
  submitted="false"
  userFormService={[Function UserFormService]}
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div
      class="form-group"
    >
      <label
        class="control-label"
        ng-reflect-translate=""
        translate=""
      >
        CIVILITY
      </label>
      <select
        class="form-control custom-select ng-untouched ng-pristine ng-invalid"
        formcontrolname="civility"
        ng-reflect-name="civility"
      >
        <option
          ng-reflect-translate=""
          translate=""
          value="0: null"
        >
          CHOOSE
        </option>
        
      </select>
    </div>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="firstname"
      ng-reflect-label="FIRST_NAME"
      ng-reflect-name="firstname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          FIRST_NAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="lastname"
      ng-reflect-label="LASTNAME"
      ng-reflect-name="lastname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LASTNAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="email"
      ng-reflect-error-text=""
      ng-reflect-label="EMAIL"
      ng-reflect-name="email"
      ng-reflect-show-error="false"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          EMAIL
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-row ng-untouched ng-pristine ng-invalid"
      formgroupname="plainPassword"
      ng-reflect-name="plainPassword"
    >
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="first"
          ng-reflect-help-text="HELP_PASSWORD_FIELD"
          ng-reflect-label="PASSWORD"
          ng-reflect-name="first"
          ng-reflect-submitted="false"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
            <small
              class="form-text text-muted"
            >
               HELP_PASSWORD_FIELD 
            </small>
          </div>
        </app-text-input>
      </div>
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-valid"
          formcontrolname="second"
          ng-reflect-error-text="FIELD_PASSWORD_MUST_BE_EQUAL"
          ng-reflect-label="PASSWORD_CONFIRMATION"
          ng-reflect-name="second"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD_CONFIRMATION
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
          </div>
        </app-text-input>
      </div>
    </div>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        disabled=""
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CREATE_ACCOUNT 
      </button>
    </div>
  </form>
</app-register-form>
`;

exports[`RegisterForm should display user exist message if provided 1`] = `
<app-register-form
  errorUserExist={[Function Boolean]}
  processing="false"
  register={[Function EventEmitter]}
  submitted="false"
  userFormService={[Function UserFormService]}
>
  <form
    class="ng-untouched ng-pristine ng-invalid"
    ng-reflect-form="[object Object]"
    novalidate=""
  >
    <div
      class="form-group"
    >
      <label
        class="control-label"
        ng-reflect-translate=""
        translate=""
      >
        CIVILITY
      </label>
      <select
        class="form-control custom-select ng-untouched ng-pristine ng-invalid"
        formcontrolname="civility"
        ng-reflect-name="civility"
      >
        <option
          ng-reflect-translate=""
          translate=""
          value="0: null"
        >
          CHOOSE
        </option>
        
      </select>
    </div>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="firstname"
      ng-reflect-label="FIRST_NAME"
      ng-reflect-name="firstname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          FIRST_NAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="lastname"
      ng-reflect-label="LASTNAME"
      ng-reflect-name="lastname"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          LASTNAME
        </label>
        <input
          class="form-control"
          placeholder=""
          type="text"
        />
        
        
        
        
        
        
      </div>
    </app-text-input>
    <app-text-input
      class="ng-untouched ng-pristine ng-invalid"
      formcontrolname="email"
      ng-reflect-error-text="USER_EXIST"
      ng-reflect-label="EMAIL"
      ng-reflect-name="email"
      ng-reflect-show-error="true"
      ng-reflect-submitted="false"
    >
      <div
        class="form-group "
      >
        <label
          class="control-label"
        >
          EMAIL
        </label>
        <input
          class="form-control is-invalid"
          placeholder=""
          type="text"
        />
        
        <div
          class="invalid-feedback custom-error"
        >
           USER_EXIST 
        </div>
        
        
        
        
        
      </div>
    </app-text-input>
    <div
      class="form-row ng-untouched ng-pristine ng-invalid"
      formgroupname="plainPassword"
      ng-reflect-name="plainPassword"
    >
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-invalid"
          formcontrolname="first"
          ng-reflect-help-text="HELP_PASSWORD_FIELD"
          ng-reflect-label="PASSWORD"
          ng-reflect-name="first"
          ng-reflect-submitted="false"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
            <small
              class="form-text text-muted"
            >
               HELP_PASSWORD_FIELD 
            </small>
          </div>
        </app-text-input>
      </div>
      <div
        class="col-md-6"
      >
        <app-text-input
          class="ng-untouched ng-pristine ng-valid"
          formcontrolname="second"
          ng-reflect-error-text="FIELD_PASSWORD_MUST_BE_EQUAL"
          ng-reflect-label="PASSWORD_CONFIRMATION"
          ng-reflect-name="second"
          ng-reflect-type="password"
          type="password"
        >
          <div
            class="form-group "
          >
            <label
              class="control-label"
            >
              PASSWORD_CONFIRMATION
            </label>
            <input
              class="form-control"
              placeholder=""
              type="password"
            />
            
            
            
            
            
            
          </div>
        </app-text-input>
      </div>
    </div>
    <div
      class="form-group"
    >
      <button
        class="btn btn-primary btn-lg btn-block"
        ng-reflect-translate=""
        translate=""
        type="submit"
      >
         CREATE_ACCOUNT 
      </button>
    </div>
  </form>
</app-register-form>
`;
